### DISCUSSION TITLE
Questions about control input

### DISCUSSION THREAD

**Discussion by QQlearning007** [alexliniger/MPCC#70] [Q&A] [answered]
hi, what is the duty cycle d ?I don't understand it. Thank you.

**Comment by alexliniger** [comment, marked as answer]
Duty cycle referees to the ratio of the high state in a PWM signal:
https://en.wikipedia.org/wiki/Pulse-width_modulation
https://en.wikipedia.org/wiki/Duty_cycle

Practically in the here used model it is kind of a mapping from the [0,1] range to a drive train force. 
`F_x = C_m1*d + Cm2*v_x*d `
In the case Cm2 = 0 you can clearly see that d just corresponds to the ratio of the full force of the motor. Cm2 models the effect that at higher motor speeds electric motors often get a bit weaker. Note that including Cm2 worked well for the tiny RC cars but in other applications we often used Cm2=0, and when necessary (such as in the case of gasoline motors) modified the constraints on d with the velocity v_x (here is an example for this idea https://www.tandfonline.com/doi/10.1080/00423114.2019.1605081 ).

Best,
Alex
